flow/flow.go

```diff
@@ -478,54 +478,42 @@ func generateRingName() string {
 	return s
 }
 
-// SetWriter adds write function to flow graph.
-// Gets flow which packets will be written to file and
-// target file name.
-// Function can panic during execution.
-func SetWriter(IN *Flow, filename string) {
-	checkFlow(IN)
-	write := makeWriter(filename, IN.current)
-	schedState.UnClonable = append(schedState.UnClonable, write)
-	IN.current = nil
-	openFlowsNumber--
-}
-
-// SetReader adds read function to flow graph.
-// Gets name of pcap formatted file and number of reads. If repcount = -1,
-// file is read infinitely in circle.
-// Returns new opened flow with read packets.
-// Function can panic during execution.
-func SetReader(filename string, repcount int32) (OUT *Flow) {
-	ring := low.CreateRing(generateRingName(), burstSize*sizeMultiplier)
-	read := makeReader(filename, ring, repcount)
-	schedState.UnClonable = append(schedState.UnClonable, read)
-	OUT = new(Flow)
-	OUT.current = ring
-	openFlowsNumber++
-	return OUT
-}
-
 // SetReceiver adds receive function to flow graph.
-// Gets port number from which packets will be received.
-// Receive queue will be added to port automatically.
-// Returns new opened flow with received packets
+// This function can get 1 or 2 arguments.
+// If argument is port number, function receive packets from this port. Receive queue added to port automatically.
+// If argument is KNI device, function receive packets from the kernel network interface.
+// If two arguments passed - it supposed to be filename (pcap formatted file) and number of readings (repcount),
+// packets are read from file. If repcount = -1, file is read infinitely in circle.
+// Returns new opened flow with received packets.
 // Function can panic during execution.
-func SetReceiver(par interface{}) (OUT *Flow) {
+func SetReceiver(par ...interface{}) (OUT *Flow) {
```

Comment:
There are three more typesafe alternatives (from simplest to more complex):
- config struct
- builder pattern (not very idiomatic in Go)
- functionl options pattern (becomes more popular lately)

[functional-options-for-friendly-apis](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) describes most options and their tradeoffs.